service/src/main/java/uk/nhs/adaptors/gp2gp/TransformJsonToXml.java

```diff
@@ -0,0 +1,225 @@
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class TransformJsonToXml {
+
+    private static final String JSON_FILE_INPUT_PATH =
+            Paths.get("src").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/input/";
+    private static final String XML_OUTPUT_PATH =
+            Paths.get("src").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/output/";
+    private static final Logger LOGGER = Logger.getLogger(TransformJsonToXml.class.getName());
+
+    private static final FhirParseService FHIR_PARSE_SERVICE = new FhirParseService();
+
+    public static void main(String[] args) throws Exception {
+
+        var inputWrapper = getFiles();
+
+        for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+            var jsonString = inputWrapper.getJsonFileInputs().get(i);
+            String xmlResult = mapJsonToXml(jsonString);
+            var fileName = inputWrapper.getJsonFileNames().get(i);
+            writeToFile(xmlResult, fileName);
+        }
+    }
+
+    private static InputWrapper getFiles() {
+
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        for (File inputJsonFile : files) {
+            LOGGER.info("Parsing File: " + inputJsonFile.getName());
+            if (FilenameUtils.getExtension(inputJsonFile.getName()).equalsIgnoreCase("json")) {
+                try {
+                    String jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + inputJsonFile.getName());
+                    jsonStringInputs.add(jsonAsString);
+                    fileNames.add(inputJsonFile.getName());
+                } catch (Exception e) {
+                    LOGGER.info("Could Not Read Json Files.");
+                }
+            } else {
+                LOGGER.info("No .json Files Found");
+            }
+        }
+        return InputWrapper.builder().jsonFileInputs(jsonStringInputs).jsonFileNames(fileNames).build();
+    }
+
+    private static String extractNhsNumber(String json) {
+        var nhsnumberSystem = "https://fhir.nhs.uk/Id/nhs-number";
+        var bundle = FHIR_PARSE_SERVICE.parseResource(json, Bundle.class);
+        var nhsNumber = bundle.getEntry().stream()
+                .map(Bundle.BundleEntryComponent::getResource)
+                .filter(resource -> ResourceType.Patient.equals(resource.getResourceType()))
+                .map(resource -> (Patient) resource)
+                .map(resource -> (Identifier) getNhsNumberIdentifier(nhsnumberSystem, resource))
+                .findFirst().get().getValue();
```

Comment:
If the stream is empty then `findFirst()` will return an empty Optional and `.get()` will throw an exception. You might prefer to write your own exception so that it's understood that that is what you intended

```suggestion
                .findFirst()
                .elseThrow(() -> new SomeKindOfException("meaningful message"))
                .getValue();
```